clients/venice-push-job/src/main/java/com/linkedin/venice/hadoop/input/avro/GenericAvroRecordReader.java

```diff
@@ -0,0 +1,82 @@
+package com.linkedin.venice.hadoop.input.avro;
+
+import com.linkedin.venice.exceptions.VeniceException;
+import com.linkedin.venice.hadoop.exceptions.VeniceSchemaFieldNotFoundException;
+import com.linkedin.venice.utils.Utils;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.NoSuchElementException;
+import org.apache.avro.Schema;
+import org.apache.avro.file.DataFileStream;
+import org.apache.avro.generic.GenericDatumReader;
+import org.apache.avro.generic.IndexedRecord;
+import org.apache.avro.mapred.AvroWrapper;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+
+/**
+ * This class reads the value field given the key field from an avro file based on the file's schema.
+ * Returns the fields as {@link Object}
+ * Currently reads only a single record containing one or multiple fields
+ */
+public class GenericAvroRecordReader {
+  private static final Logger LOGGER = LogManager.getLogger(GenericAvroRecordReader.class);
+  private final InputStream inputStream;
+  private final DataFileStream avroDataFileStream;
+  private final Schema fileSchema;
+  private final Object avroObject;
+
+  public GenericAvroRecordReader(FileSystem fs, String file) {
+    if (fs != null && file != null) {
```

Comment:
nit: we can use `Validate.notNull` for these.

Reply:
Removed this file itself, so skipping this. Thanks for the comment.

Reply:
Used this inside `ValidateSchemaAndBuildDictMapperOutputReader`. Thanks for the suggestion.